### ISSUE TITLE
Stuck with enabling/disabling addon "dns"

### ISSUE THREAD

**Issue by crustulumtheoneandonly** [canonical/microk8s#3346]
Following situation forced me to completly reinstall microk8s on all nodes multiple times:
(in the following text addon=plugin)

If there is a update or any network related change (hostname etc...) and the microk8s cluster dns doesnt work properly anymore, I tried to disable and enable the dns plugin.

Unfortunatly, if the script ("microk8s disable dns) fails when killing a pod or cant reach a dead node the script will stop, but the dns plugin will still stay status "enabled". When I try to run the "microk8s disable dns" again, the error "no deployment coredns found" comes up. When I try to "microk8s enable dns", the error "dns already enabled" comes up. So there Im stuck at.

Could it maybe possible to "force" / "purge" remove the dns plugin a second time or force enable / reset it?

Maybe there is a even simpler solution to the problem?

Everything else is great work! Keep it up :)

**Comment by ktsakalozos** [comment]
Hi @crustulumtheoneandonly, thank you for reporting this. We are looking for ways to improve the robustness of the addons framework.

Changing the properties of a node (eg its hostname) often results in the k8s cluster considering that specific node a new one so the safest thing to do in such cases is to first remove the node, reconfigure it while it is outside the cluster and then re-add it as if it was a new one.

When you find yourself with an addon that does not get enabled/disabled cleanly you could consider applying it manually. For example the enable dns script [1] applies the coredns manifest and then attempts to configure kubelet in all nodes. 

[1] https://github.com/canonical/microk8s-core-addons/blob/main/addons/dns/enable

**Comment by crustulumtheoneandonly** [comment]
Hi @ktsakalozos thanks for the quick response!

Im just wondering now, if everytime I remove or add a node:
Do I need to re-apply the dns script to configure the kubelet of the new node?

I have a cluster set up in a wireguard mesh vpn network. I made several attempts to get the dns problematic stable, my current configuration was in following order deployed:

1. install and start microk8s on single node
2. enable dns on this single-node cluster
3. join all the other nodes

As far as I noticed, the cluster is stable for a few days now.
Is the cluster dns config dependent on this one node or do I need to re-apply the dns addon script to configure the other master nodes?

Sorry for the greenhorn questions, Im pretty new to microk8s and kubernetes but its really awesome!
